src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTargetRHelper.java

```diff
@@ -36,13 +36,17 @@ void initialize(EventBus eventBus)
       eventBus_ = eventBus;
    }
 
-   public void customSource(EditingTarget editingTarget)
+   public boolean customSource(EditingTarget editingTarget)
    {
       TextEditingTargetCommentHeaderHelper customSource = new TextEditingTargetCommentHeaderHelper(
          docDisplay_.getCode(),
          "source",
          "#"
       );
+      
+      // No work to do if we didn't find a comment header
+      if (!customSource.hasCommentHeader())
```

Comment:
Looks like we also need this fix in: https://github.com/rstudio/rstudio/search?q=previewSource.buildCommand&unscoped_q=previewSource.buildCommand, was able to reproduce with D3 and SQL files as well.

Reply:
Done!